### ISSUE TITLE
vulnerability in future

### ISSUE THREAD

**Issue by spereverde** [plone/plone-backend#113]
Not sure this is the right place.
Our jenkins ppelines perform automatic securitiny vulnerability checks and when I tried to build a docker image based on plone-backend 6.0.2, it gives the following warning:
```
09:32:14  [2023-03-08T08:32:14.876Z] CVE-2022-40899 (HIGH) for future:0.18.2:/app/lib/python3.11/site-packages/future-0.18.2.dist-info/METADATA - python-future: remote attackers can cause denial of service via crafted Set-Cookie header from malicious web server: (https://avd.aquasec.com/nvd/cve-2022-40899)
09:32:14  [2023-03-08T08:32:14.876Z]     An issue discovered in Python Charmers Future 0.18.2 and earlier allows remote attackers to cause a denial of service via crafted Set-Cookie header from malicious web server.
09:32:14  [2023-03-08T08:32:14.876Z] 
09:32:14  [2023-03-08T08:32:14.876Z] Fixed in 0.18.3
09:32:14  [2023-03-08T08:32:14.876Z] -----
```
I guess we should update future in https://dist.plone.org/release/6.0.2/constraints.txt
If anyone directs me to where I can do that, I'll make a PR

**Comment by thet** [comment]
Hi, tnx for the heads up!

Seems future is not used in `plone-backend` directly, but in some other Plone core packages like diazo, plone.testing and so. We should get rid of this dependency because with Modern Python 3 that isn't necessary anymore.

However, to fix that you'd need to make PR in `buildout.coredev` in the branches 6.0, 6.1 and 5.2.
You find the version pin in `release/constraints.txt`. Looks like the top level constraints.txt is generated:

```
dev/plone/buildout.coredev-60$ rg future
[...]
release/constraints.txt
115:future==0.18.2
```

**Comment by yurj** [comment]
"I've created a PR which removes dependency on future in diazo: [Replace future with six by petschki · Pull Request #85 · plone/diazo · GitHub 2](https://github.com/plone/diazo/pull/85) ... inspecting the dependencies with plone.versioncheck there are Products.BTreeFolder2 and repoze.xmliter which also depend on future but don't use it in the code anymore. I think it would be fine to remove this dependency there."

https://community.plone.org/t/plone-still-depends-on-the-future-python2-3-python-egg/15779/3

**Comment by fredvd** [comment]
@spereverde Clossing this issue as Plone 6.0.3 has been released which has an updated future 0.18.3.   The dependency in Diazo has been removed, as well as in Products.Btree2Folder.  (https://github.com/zopefoundation/Products.BTreeFolder2/issues/16)

but it's still present in repoze.xmliter: I've created an issue there.   https://github.com/repoze/repoze.xmliter/issues/10

Closing this issue
